### PR TITLE
Core - Add debug information for reserved variable error

### PR DESCRIPTION
**Pull request by jonpas**
**When merged this pull request will:**
- Debug for #638

### CODE REVIEW

File: addons/sys_core/fnc_processRadioSpeaker.sqf [hunk 1]
```diff
@@ -27,6 +27,13 @@ TRACE_2("",_unit,_playerRadios);
 
 private _radioId = _unit getVariable [QGVAR(currentSpeakingRadio), ""];
 if (_radioId == "") exitWith { false };
+
+// Debug #638
+if (isNil "_radioId") exitWith {
+    WARNING_1("reserved variable debug 1 - report on GitHub!",_radioId);
```
Comment (PabstMirror) [inline addons/sys_core/fnc_processRadioSpeaker.sqf]:
```suggestion
    WARNING_1("reserved variable debug 1 - report on GitHub! [%1]",_radioId);
```

File: addons/sys_core/fnc_processRadioSpeaker.sqf [hunk 2]
```diff
@@ -43,6 +50,12 @@ if (!GVAR(speaking_cache_valid)) then {
     #endif
     _okRadios = (_okRadios select 0) select 1;
 
+    // Debug #638
+    if (isNil "_radioId") exitWith {
+        WARNING_3("reserved variable debug 2 - report on GitHub!",_radioId,_okRadios,_playerRadios);
```
Comment (PabstMirror) [inline addons/sys_core/fnc_processRadioSpeaker.sqf]:
```suggestion
        WARNING_3("reserved variable debug 2 - report on GitHub! [%1-%2-%3]",_radioId,_okRadios,_playerRadios);
```

**Comment by jonpas** [comment]
I always forget this is not the same as `TRACE_n`.

File: addons/sys_core/fnc_processRadioSpeaker.sqf [hunk 3]
```diff
@@ -43,6 +50,12 @@ if (!GVAR(speaking_cache_valid)) then {
     #endif
     _okRadios = (_okRadios select 0) select 1;
 
+    // Debug #638
+    if (isNil "_radioId") exitWith {
```
Comment (PabstMirror) [inline addons/sys_core/fnc_processRadioSpeaker.sqf:54]: Should this nil check be on `_okRadios`? _radioId can't change from the last check
also exitWith return value will be lost to scope

Comment (jonpas) [reply addons/sys_core/fnc_processRadioSpeaker.sqf:54]: That's what I am trying to figure out, somehow `_radioId` is `nil` in the below `HASH_SET` operation, even though it should be `""` due to `getVariable` default value on `nil`.

Does `setVariable` also throw "reserved variable" error if you try to assign nil to it? I don't think so.

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**